### ISSUE TITLE
Calling partial() as a function to check for the existence of a partial fully executes the partial

### ISSUE THREAD

**Issue by LukeTowers** [wintercms/winter#263]
Calling `partial(path)` as [documented to determine if `path` exists](https://wintercms.com/docs/markup/tag-partial) and can be run as a partial fully executes the partial which includes any `put` statements (i.e. `put scripts`) which can lead to further issues like script resources added via `put scripts` being duplicated in the final output. See https://github.com/OFFLINE-GmbH/oc-microcart-plugin/blob/master/components/cart/payment_data.htm#L3 for an example of an affected plugin.

**Comment by bennothommo** [comment]
@LukeTowers while we could certainly change the behaviour of the `partial()` method, it might be being used (somewhat incorrectly) to load partials and therefore could be a breaking change.

I think we might have to, in this instance, create a new method (perhaps `exists()`) that checks existence and update the docs to indicate the true behaviour of the partial method.

**Comment by LukeTowers** [comment]
@bennothommo yeah, that's what I was thinking, something like `partialExists()`, `pageExists()`, etc
